CIP-JonathanKnowles-CoinSelectionAlgorithms/CIP-JonathanKnowles-CoinSelectionAlgorithms.md

```diff
@@ -0,0 +1,983 @@
+---
+CIP: ?
+Title: Coin Selection Algorithms for Cardano
+Author: Jonathan Knowles <jonathan.knowles@iohk.io>
+Comments-URI: ?
+Status: Draft
+Type: Informational
+Created: 2020-05-04
+License: CC-BY-4.0
+---
```

Comment:
This header is in the style of [RFC 822](https://www.ietf.org/rfc/rfc822.txt), as [specified](https://github.com/cardano-foundation/CIPs/blob/master/CIP1/CIP1.md#cip-header-preamble) by [CIP 1](https://github.com/cardano-foundation/CIPs/blob/master/CIP1/CIP1.md).

Though perhaps using a _pre-formatted block_ here would be more readable, in the same way as is done for Bitcoin Improvement Proposals (for example, see [BIP 100](https://en.bitcoin.it/wiki/BIP_0100)).

#### Justification

The GitHub markdown renderer renders RFC 822-style preambles as very wide tables with one column per field. For example:

![horizontal-table](https://user-images.githubusercontent.com/206319/81145726-6da50e00-8fa9-11ea-92f0-db027a248133.png)

For readers without wide monitors, this might result in important information being initially hidden from view. Of course, users can always scroll horizontally, but many consider horizontal scrolling to be [bad for accessibility](https://www.nngroup.com/articles/horizontal-scrolling/).

Compare this to the pre-formatted block style used with Bitcoin Improvement Proposals:
```  BIP: 100
  Layer: Consensus (hard fork)
  Title: Dynamic maximum block size by miner vote
  Author: Jeff Garzik <jgarzik@gmail.com>
          Tom Harding <tomh@thinlink.com>
          Dagur Valberg Johannsson <dagurval@pvv.ntnu.no>
  Comments-Summary: No comments yet.
  Comments-URI: https://github.com/bitcoin/bips/wiki/Comments:BIP-0100
  Status: Rejected
  Type: Standards Track
  Created: 2015-06-11
  License: BSD-2-Clause
```
Now, as more information is added, the content scales vertically.